dashboard/app/controllers/reports_controller.rb

```diff
@@ -29,6 +30,28 @@ def header_stats
     render file: 'shared/_user_stats', layout: false, locals: {user: current_user}
   end
 
+  def user_progress
```

Comment:
This should now be the only consumer of header_progress.  Future work will yank that helper into this API, and then break it apart into its static & dynamic parts.